pkg/storage/stores/composite_store_entry.go

```diff
@@ -90,7 +90,10 @@ func (c *storeEntry) GetChunks(
 func filterForTimeRange(refs []*logproto.ChunkRef, from, through model.Time) []chunk.Chunk {
 	filtered := make([]chunk.Chunk, 0, len(refs))
 	for _, ref := range refs {
-		if through >= ref.From && from < ref.Through {
+		// Only include chunks where the query start time (from) is < the chunk end time (ref.Through)
+		// and the query end time (through) is >= the chunk start time (ref.From)
+		// A special case also exists where a chunk can contain a single log line which results in ref.From being equal to ref.Through, and that is equal to the from time.
+		if (through >= ref.From && from < ref.Through) || (ref.From == from && ref.Through == from) {
```

Comment:
do you know if there's a relevant difference between using `through.After(x)` vs `through > x`?

Reply:
I don't think so. `through.After()` is an additional function call, but that might get inlined anyway.

Reply:
Quickly checked the following example:


```go
package main

import (
	"fmt"
)

type Time int64

func (t Time) After(other Time) bool {
	return t > other
}

func main() {
	a := Time(1000)
	fmt.Println(a.After(500))
}

```


```console
$ go build -gcflags -m inline.go 
# command-line-arguments
./inline.go:9:6: can inline Time.After
./inline.go:15:21: inlining call to Time.After
./inline.go:15:13: inlining call to fmt.Println
<autogenerated>:1: inlining call to Time.After
./inline.go:15:13: ... argument does not escape
./inline.go:15:21: ~r0 escapes to heap
```